src/prognostic_equations/edmfx_closures.jl

```diff
@@ -281,43 +281,126 @@ function mixing_length(
 end
 
 """
-    turbulent_prandtl_number(params, obukhov_length, ᶜRi_grad)
+    turbulent_prandtl_number(params, ᶜlinear_buoygrad, ᶜstrain_rate_norm)
 
 where:
-- `params`: set with model parameters
-- `obukhov_length`: surface Monin Obukhov length
-- `ᶜRi_grad`: gradient Richardson number
+- `params`: Parameters set
+- `ᶜlinear_buoygrad`: N^2, Brunt-Väisälä frequency squared [1/s^2].
+- `ᶜstrain_rate_norm`: Frobenius norm of strain rate tensor, |S| [1/s].
+
+Returns the turbulent Prandtl number based on the gradient Richardson number.
+
+The formula implemented is from Li et al. (JAS 2015, DOI: 10.1175/JAS-D-14-0335.1, their Eq. 39),
+with a reformulation and correction of an algebraic error in their expression:
+
+    Pr_t(Ri) = (X + sqrt(max(X^2 - 4*Pr_n*Ri, 0))) / 2
 
-Returns the turbulent Prandtl number give the obukhov length sign and
-the gradient Richardson number, which is calculated from the linearized
-buoyancy gradient and shear production.
+where X = Pr_n + ω_pr * Ri and Ri = min( N^2 / max(2*|S|, eps), Ri_max )
+using parameters Pr_n = Prandtl_number_0, ω_pr = Prandtl_number_scale, and Ri_max.
+This formula applies in both stable (Ri > 0) and unstable (Ri < 0) conditions.
 """
-function turbulent_prandtl_number(
-    params,
-    obukhov_length,
-    ᶜlinear_buoygrad,
-    ᶜstrain_rate_norm,
-)
+function turbulent_prandtl_number(params, ᶜlinear_buoygrad, ᶜstrain_rate_norm)
+    FT = eltype(params)
+    turbconv_params = CAP.turbconv_params(params)
+    eps_FT = eps(FT)
+
+    # Parameters from CliMAParams
+    Pr_n = CAP.Prandtl_number_0(turbconv_params) # Neutral Prandtl number
+    ω_pr = CAP.Prandtl_number_scale(turbconv_params) # Prandtl number scale coefficient
+    # TODO The default value for this is 0.25. Testing if 100 works, so I can update ClimaParams
+    #Ri_max = CAP.Ri_max(turbconv_params) # Max Richardson number limit
+    Ri_max = FT(0.25)
```

Comment:
Please replace Ri_max parameter by Pr_max (which is really what we want to limit).